my_map.rb

```diff
@@ -0,0 +1,64 @@
+class Array
+  def my_map
+    array = []
+    if block_given?
+      self.each { |value| array << yield(value) }
+      array
+    else
+      self.to_enum(:map)
+    end
+  end
+  def my_map!
+    if block_given?
+     self.my_map.with_index { |value, i| self[i] = yield(value) }
+    else
+      self.to_enum(:map)
+    end
+  end
+  def my_reduce(sum = nil, symbol = nil)
+    if sum.nil? && symbol.nil?
+      sum = self.first
+      index = 0
+      self.each do |n|
+        if index != 0
```

Comment:
`Array.shift` can help to remove  `if index != 0`, but be careful with the initial array.

Reply:
fixed